localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp

```diff
@@ -280,6 +280,22 @@ void NDTScanMatcher::timer_diagnostic()
       diag_status_msg.level = diagnostic_msgs::msg::DiagnosticStatus::ERROR;
       diag_status_msg.message += "skipping_publish_num exceed limit. ";
     }
+    if (
+      state_ptr_->count("nearest_voxel_transformation_likelihood") &&
+      std::stod((*state_ptr_)["nearest_voxel_transformation_likelihood"]) < 2.3) {
+      diag_status_msg.level = diagnostic_msgs::msg::DiagnosticStatus::WARN;
+      diag_status_msg.message += "NDT score is unreliably low. ";
+    }
+    if (
+      state_ptr_->count("exe_time") && std::stod((*state_ptr_)["exe_time"]) > 13 &&
+      std::stod((*state_ptr_)["exe_time"]) < 24) {
+      diag_status_msg.level = diagnostic_msgs::msg::DiagnosticStatus::WARN;
+      diag_status_msg.message += "NDT exe time is increasing. ";
+    }
+    if (state_ptr_->count("exe_time") && std::stod((*state_ptr_)["exe_time"]) >= 24) {
```

Comment:
(Sorry for adding comments little by little :pray: )

Since publishing the ERROR state have a large influence on the whole system, especially to the system_error_monitor and its followers, how about removing the ERROR state for now and only introduce WARN state?

We can see how it performs (e.g. checking its false positives) and gather information, and then decide to add ERROR state or not based on the data.

Reply:
I have no problem with that. On the contrary, I am very pleased with your opinions. Surely I can do that. I'm also not sure switching ERROR state by exe_time. Then I wil just keep 24 as default and only parameter for switching WARN state by exe_time.

Reply:
Thank you! Let me know once it's done.